### PR TITLE
ShareLink: Apple Music: Support user-created playlists

### PR DESCRIPTION
**Pull request by mfrister**
User-created playlists have a slightly different URL format, I've extended the regexp to support them as well.

### CODE REVIEW

**Comment by pwt** [comment]
@mfrister: Thanks for submitting this PR.
@amelchio : I wonder if you'd be able to give this a quick review / test?

File: soco/plugins/sharelink.py [hunk 1]
```diff
@@ -153,9 +153,12 @@ def canonical_uri(self, uri):
         if match:
             return "album:" + match.group(1)
 
+        # Apple-created playlist
         # https://music.apple.com/dk/playlist/power-ballads-essentials/pl.92e04ee75ed64804b9df468b5f45a161
+        # User-created playlist
+        # https://music.apple.com/de/playlist/unnamed-playlist/pl.u-rR2PCrLdLJk
         match = re.search(
-            r"https://music\.apple\.com/\w+/playlist/[^/]+/(pl\.[a-f\d]+)", uri
+            r"https://music\.apple\.com/\w+/playlist/[^/]+/(pl\.[a-zA-Z\-0-9\d]+)", uri
```
Comment (amelchio) [inline soco/plugins/sharelink.py:161]:
```suggestion
            r"https://music\.apple\.com/\w+/playlist/[^/]+/(pl\.[-a-zA-Z0-9]+)", uri
```

**Review by amelchio — APPROVED** [review]
Thanks! I can confirm the issue and have tested that the fix works 👍.

The regex can be simplified a bit.

**Comment by pwt** [comment]
@amelchio Thanks for testing this and for the regex simplification.

**Comment by pwt** [comment]
I've now released v0.26.2 containing this improvement. Thanks again for your contributions.

**Pull request merged**